File: method_chain_pattern/mcp.py
class Calculator :
  def __init__(self):
    self.result = 0

  def add(self, value):
    self.result += value
    return self
  def subtract(self, value):
    self.result -= value
    return self
  def multiply(self, value):
    self.result *= value
    return self
  def divide(self, value):
    if value != 0:
      self.result /= value
    else:
      print("Cannot divide by zero")
    return self
  def get_result(self):
    return self.result
  

# Example usage
calculator = Calculator()
result = (calculator.add(10)
           .subtract(5)
           .multiply(2)
           .divide(3)
           .get_result())
print("Result:", result)  

File: method_chain_pattern/mcp.cs
using System;

public class Calculator
{
  private double result;

  public Calculator()
  {
    result = 0;
  }

  public Calculator Add(double value)
  {
    result += value;
    return this;
  }

  public Calculator Subtract(double value)
  {
    result -= value;
    return this;
  }

  public Calculator Multiply(double value)
  {
    result *= value;
    return this;
  }

  public Calculator Divide(double value)
  {
    if (value != 0)
    {
      result /= value;
    }
    else
    {
      Console.WriteLine("Cannot divide by zero");
    }
    return this;
  }

  public double GetResult()
  {
    return result;
  }
}

public class Program
{
  public static void Main(string[] args)
  {
    Calculator calculator = new Calculator();
    double result = calculator.Add(10).Subtract(2).Multiply(3).Divide(4).GetResult();
    Console.WriteLine("The result is: " + result);
  }
}
